django/db/backends/postgresql/base.py

```diff
@@ -469,6 +484,23 @@ class ServerBindingCursor(CursorMixin, Database.Cursor):
     class Cursor(CursorMixin, Database.ClientCursor):
         pass
 
+    class ServerSideCursor(
```

Comment:
Thinking about it I feel less and less happy about this new class (and our `ServerBindingCursor` for that matter). As it looks there is no longer a simple way for code using cursors to check for the cursor type -- Django has now three of them :D Sure you can check for `CursorMixin` but that feels wrong as well. I guess we won't find an easy way to make our `Cursor` class a base for `Server*Cursor`?

Reply:
I'm not a fan of how things are organized given we're somewhat at the mercy of changes to the internal `ClientCursorMixin` and the very changes that would make our changes easier but that's was the less invasive solution I could come up after a few hours invested in this adventure.

If you can think of a better way to organize things in a context where we'll need to deal with async variants in a few months from now I'd be curious to see what you can come up with.

> As it looks there is no longer a simple way for code using cursors to check for the cursor type

Is that really a concern given we usually wrap them anyway when using `django.db.Connection.cursor`?

> I guess we won't find an easy way to make our Cursor class a base for Server*Cursor?

We could make `ServerSideCursor(Cursor, Database.ServerCursor)` and things would work, not for `ServerBindingCursor` though. I guess we could have

- `Cursor(CursorMixin, Database.Cursor)`
- `ServerBindingCursor = Cursor`; an alias
- `ClientBindingCursor(Cursor, Database.ClientCursor)`
- `ServerSideClientBindingCursor(ClientBindingCursor, Database.ServerCursor)`